towsg84 and tomercator unit [Improve helpers.dart with better documentation and code style]

Newmanjack opened the pull request:
Added toWGS84 and toMercator as member functions of the coordinate types in the helper file 


- Update documentation to use proper Dart doc syntax with types in []
- Convert var to final where appropriate
- Maintain consistent documentation style across the codebase
- Improve readability of function parameter documentation

Newmanjack committed:
Add pointOnFeature functionality to find points on GeoJSON features

Newmanjack committed:
Update pointOnFeature tests and export functionality in turf.dart

Newmanjack committed:
Move point_on_feature test to components directory and improve test organization

Newmanjack committed:
Add point_on_feature benchmark for performance testing

Newmanjack committed:
Added toWGS84 and toMercator as member functions of the coordinate types

Newmanjack committed:
Improve documentation for coordinate projection functions to follow Dart standards

Newmanjack committed:
Add comprehensive helpers examples with coordinate system conversions

Newmanjack committed:
Improve helpers.dart with better documentation and code style

- Update documentation to use proper Dart doc syntax with types in []
- Convert var to final where appropriate
- Maintain consistent documentation style across the codebase
- Improve readability of function parameter documentation

Newmanjack committed:
Remove point_on_feature files from this branch

- Removed lib/src/point_on_feature.dart
- Removed lib/point_on_feature.dart
- Removed benchmark/point_on_feature_benchmark.dart
- Removed test/components/point_on_feature_test.dart

These files are now part of the point_on_feature_final branch and pull request

lukas-h commented on lib/src/helpers.dart:
```diff
@@ -180,3 +206,83 @@ num convertArea(num area,
+///
+/// Valid systems: [CoordinateSystem.wgs84], [CoordinateSystem.mercator]
+/// Returns: [List] of coordinates in the target system
+List<double> convertCoordinates(
```
we have this special type `Position` that we use for coordinates (that implements a few extra utilities (like some maths) for coordinates. Can we use this instead of `List<double>`

lukas-h commented on lib/src/helpers.dart:
```diff
@@ -180,3 +206,83 @@ num convertArea(num area,
+/// Valid systems: [CoordinateSystem.wgs84], [CoordinateSystem.mercator]
+/// Returns: [List] of coordinates in the target system
+List<double> convertCoordinates(
+  List<num> coord, 
```
here too

lukas-h commented on lib/src/helpers.dart:
```diff
@@ -180,3 +206,83 @@ num convertArea(num area,
+///
+/// Valid inputs: [List] of [longitude, latitude]
+/// Returns: [List] of [x, y] coordinates in meters
+List<double> toMercator(List<num> coord) {
```
`Position`

lukas-h commented on lib/src/helpers.dart:
```diff
@@ -180,3 +206,83 @@ num convertArea(num area,
+///
+/// Valid inputs: [List] of [x, y] in meters
+/// Returns: [List] of [longitude, latitude] coordinates
+List<double> toWGS84(List<num> coord) {
```
`Position`

lukas-h commented on lib/src/helpers.dart:
```diff
@@ -180,3 +206,83 @@ num convertArea(num area,
+  // Clamp latitude to valid range
+  final clampedLatitude = max(min(latitude, 90.0), -90.0);
+  
+  return [longitude, clampedLatitude];
```
one important thing when converting to `Position` return type is that we support a third parameter after lng and lat which is altitude (alt).

lukas-h commented on lib/src/helpers.dart:
```diff
@@ -180,3 +206,83 @@ num convertArea(num area,
+  final clampedX = max(min(x, mercatorLimit), -mercatorLimit);
+  final clampedY = max(min(y, mercatorLimit), -mercatorLimit);
+  
+  return [clampedX, clampedY];
```
one important thing when converting to Position return type is that we support a third parameter after lng and lat which is altitude (alt).

Newmanjack committed:
Refactor coordinate system conversion functions to use Position type

- Updated convertCoordinates, toMercator, and toWGS84 functions to use Position type instead of List
- Added proper altitude preservation during coordinate conversions
- Improved error handling with ArgumentError and descriptive messages
- Added tests to verify conversions and altitude preservation
- Updated Progress.md to mark functions as implemented

Newmanjack commented:
Just want to confirm - 

Did you want me to merge the testing file "helpers.test"  into meta.test as that would be best practice? 

and therefore add it onto the next branch that requires it and continue adding it on?

As  @aardrop mentioned in the discord?

hamishdgx commented:
@lukas-h I think we can close this PR. It looks like its been continued in https://github.com/dartclub/turf_dart/pull/219.

lukas-h commented:
> @lukas-h I think we can close this PR. It looks like its been continued in #219.

sure!
